test/NuGet.Core.Tests/NuGet.Packaging.Test/NupkgMetadataFileFormatTests.cs

```diff
@@ -63,58 +54,39 @@ public void Read_V2FileContents_ReturnsCorrectObject()
         public void Write_WithObject_RoundTrips()
         {
             var nupkgMetadataFileContent = @"{
-                ""version"": 2,
-                ""contentHash"": ""NhfNp80eWq5ms7fMrjuRqpwhL1H56IVzXF9d+OIDcEfQ92m1DyE0c+ufUE1ogB09+sYLd58IO4eJ8jyn7AifbA=="",
-                ""source"": ""https://source/v3/index.json""
-            }";
+  ""version"": 2,
+  ""contentHash"": ""NhfNp80eWq5ms7fMrjuRqpwhL1H56IVzXF9d+OIDcEfQ92m1DyE0c+ufUE1ogB09+sYLd58IO4eJ8jyn7AifbA=="",
+  ""source"": ""https://source/v3/index.json""
+}";
 
-            NupkgMetadataFile metadataFile = null;
+            NupkgMetadataFile metadataFile = Read(nupkgMetadataFileContent, NullLogger.Instance, string.Empty);
 
-            using (var reader = new StringReader(nupkgMetadataFileContent))
+            using (var stream = new MemoryStream())
             {
-                metadataFile = NupkgMetadataFileFormat.Read(reader, NullLogger.Instance, string.Empty);
-            }
-
-            using (var writer = new StringWriter())
-            {
-                NupkgMetadataFileFormat.Write(writer, metadataFile);
-                var output = JObject.Parse(writer.ToString());
-                var expected = JObject.Parse(nupkgMetadataFileContent);
-
-                Assert.Equal(expected.ToString(), output.ToString());
+                NupkgMetadataFileFormat.Write(stream, metadataFile);
+                stream.Position = 0;
+                string content;
+                using (var reader = new StreamReader(stream))
+                {
+                    content = reader.ReadToEnd();
+                }
+
+                Assert.Equal(nupkgMetadataFileContent, content);
             }
         }
 
         [Theory]
         [InlineData("")]
         [InlineData("1")]
         [InlineData("[]")]
-        public void Read_ContentsNotAnObject_ThrowsException(string contents)
-        {
-            // Arrange
-            var logger = new TestLogger();
-
-            // Act
-            using (var stringReader = new StringReader(contents))
-            {
-                Assert.Throws<InvalidDataException>(() => NupkgMetadataFileFormat.Read(stringReader, logger, "from memory"));
-            }
-
-            // Assert
-            Assert.Equal(1, logger.Messages.Count);
-        }
-
-        [Fact]
-        public void Read_ContentsInvalidJson_ThrowsJsonReaderException()
```

Comment:
This test exposes a change in behaviour. Previously, it would throw a Newtonsoft.Json.JsonReaderException. But this is arguably a leaky abstraction, and it's not reasonable to maintain given we're migrating off Newtonsoft.Json.  I thought `InvalidDataException` is better, so I merged this test in with the other invalid data test.